provider/cmd/crd2pulumi/gen/generate.go

```diff
@@ -130,87 +128,86 @@ func (gen *CustomResourceGenerator) Name() string {
 	return gen.Plural + "." + gen.Group
 }
 
-// GetVersionNames returns a slice of the versions supported by this CRD.
-func (gen *CustomResourceGenerator) GetVersionNames() []string {
+// VersionKeys returns a slice of the versions supported by this CRD.
+func (gen *CustomResourceGenerator) VersionKeys() []string {
 	versionNames := make([]string, 0, len(gen.Versions))
 	for versionName := range gen.Versions {
 		versionNames = append(versionNames, versionName)
 	}
 	return versionNames
 }
 
-func (gen *CustomResourceGenerator) GenerateCode() error {
-	objectTypeSpecs := GetObjectTypeSpecs(gen.Versions, gen.Name(), gen.Kind)
+// VersionNames returns a slice of the full names of each version, in the format
+// <plural>.<group>/<version>.
+func (gen *CustomResourceGenerator) VersionNames() []string {
+	versions := gen.VersionKeys()
+	name := gen.Name()
+	for i, version := range versions {
+		versions[i] = name + "/" + version
+	}
+	return versions
+}
+
+// getVersion returns the <version> field of a string in the format
+// <plural>.<group>/<version>
+func getVersion(versionName string) string {
+	return strings.Split(versionName, "/")[1]
+}
 
+// Generate outputs strongly-typed args for the CustomResourceGenerator's
+// CRD in the target language and output folder
+func (gen *CustomResourceGenerator) Generate() error {
 	switch gen.Language {
 	case NodeJS:
-		types, err := nodejs.GenerateTypes(objectTypeSpecs)
+		buffer, err := gen.genNodeJS()
 		if err != nil {
-			return fmt.Errorf("generate code %v", err)
+			return errors.Wrapf(err, "generating nodeJS code")
 		}
-		classes := gen.GenerateNodeJSClasses()
 
-		file, err := os.Create(gen.OutputPath)
+		outputFile := filepath.Join(gen.OutputDir, gen.Plural+".ts")
+		file, err := os.Create(outputFile)
 		if err != nil {
-			return fmt.Errorf("creating file at %s: %v", gen.OutputPath, err)
+			return errors.Wrapf(err, "creating file %s", outputFile)
 		}
 		defer file.Close()
-		file.WriteString(types)
-		file.WriteString(classes)
-		return nil
+
+		_, err = buffer.WriteTo(file)
+		if err != nil {
+			return errors.Wrapf(err, "writing to %s", outputFile)
+		}
+	case Go:
+		buffers, err := gen.genGo()
+		if err != nil {
+			return errors.Wrapf(err, "generating Go code")
+		}
+
+		for versionName, buffer := range buffers {
+			packageDir := filepath.Join(gen.OutputDir, getVersion(versionName))
+			if _, err := os.Stat(packageDir); os.IsNotExist(err) {
+				err = os.Mkdir(packageDir, 0755)
+				if err != nil {
+					return errors.Wrapf(err, "creating directory %s", packageDir)
+				}
+			}
```

Comment:
Using `MkdirAll` would be more robust in case the path includes a nested directory:

```go
err := os.MkdirAll(packageDir, 0755)
if err != nil {
    return errors.Wrapf(err, "creating directory %s", packageDir)
}
```